internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go

```diff
@@ -115,7 +136,10 @@ func (storage *secureStorage) Delete(ctx context.Context, key string) error {
 }
 
 func (storage *secureStorage) Exists(ctx context.Context, key string) bool {
-	_, err := storage.Load(ctx, key)
+	name := storage.constructSecretName(key)
```

Comment:
What's the motivation of the change in the Exists method?

Reply:
Good catch on this: the change is desirable but not critical for this PR i.e it could be a separate PR.

Exists should check if the key truly exists in the backend storage. With the use of `.Load` which has caching (which needs to be revisited) built on it - we were in some some cases returning `true` since the content is in the cache but the certificate isn't in the storage backend.

Reply:
taken out of the PR